<launch>

  <!-- <include file="$(find unsw_vision_utils)/launch/realsense.launch"/> -->
  <!-- <include file="$(find unsw_vision_utils)/launch/rs_d435_camera_with_model.launch"/> -->

  <!-- <include file="$(find hsrb_rosnav_config)/launch/hsrb_nav.launch"/> -->

  <!-- <node pkg="unsw_vision_utils" type="republisher" name="republisher" output="screen">
    <param name="input_rgb_topic" value="/hsrb/head_rgbd_sensor/rgb/image_rect_color"/>
    <param name="input_pcd_topic" value="/hsrb/head_rgbd_sensor/depth_registered/rectified_points"/> -->
    <!-- <param name="input_rgb_topic" value="/camera/color/image_raw"/>
    <param name="input_pcd_topic" value="/camera/depth/color/points"/>   -->
    <!-- <param name="output_rgb_topic" value="/unsw_vision/republisher/rgb"/>
    <param name="output_pcd_topic" value="/unsw_vision/republisher/pcd"/>        
    <param name="use_compressed" value="false"/>
    <param name="use_exact" value="false"/>
    <param name="framerate" value="5"/>
  </node>  -->

  <include file="$(find unsw_vision_utils)/launch/pcl-republisher.launch"/>

  




  <!-- Core Process - Object/Person Detection -->
  <include file="$(find yolov7_ros)/launch/yolov7.launch"/>
  


<node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen"/>
  <!-- PositionGenerator nodelet -->
  <node pkg="nodelet" type="nodelet" name="PositionGenerator_nodelet" args="load unsw_vision_utils/PositionGenerator nodelet_manager">
    <param name="output_topic" type="str" value="/unsw_vision/detections/objects/positions"/>
    <param name="input_pcd_topic" type="str" value="/points"/>
    <param name="input_obj_topic" type="str" value="/unsw_vision/detections/objectsWithBagHandle"/>
    <param name="target_frame_id" type="str" value="map"/>
    <param name="source_frame_id" type="str" value="head_rgbd_sensor_rgb_frame"/>
  </node>


<node pkg="nodelet" type="nodelet" name="BagHandleDetector_nodelet" args="load unsw_vision_utils/BagHandleDetector nodelet_manager">
    <param name="in_topic" type="str" value="/unsw_vision/detections/objects"/>
    <param name="out_topic" type="str" value="/unsw_vision/detections/objectsWithBagHandle"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="PersonPointingDetector_nodelet" args="load unsw_vision_utils/PersonPointingDetector nodelet_manager">
    <param name="input_topic" type="str" value="/unsw_vision/detections/objects/positions"/>
    <param name="output_topic" type="str" value="/unsw_vision/detections/personPointing"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="BagScanner_nodelet" args="load unsw_vision_utils/BagScanner nodelet_manager">
    <param name="in_topic" type="str" value="/unsw_vision/detections/objects/positions"/>
    <param name="out_topic" type="str" value="/unsw_vision/detections/pointedBag"/>
    <param name="point_topic" type="str" value="/unsw_vision/detections/personPointing"/>
    <param name="ready_topic" type="str" value="/bag_scan_ready"/>
  </node>

</launch>
